### PR TITLE
lost equip sound bug

### PR DESCRIPTION
**Pull request by tcm390**
This issue is caused by that `isLocalPlayer` hasn't been set in character-sfx constructor (undefined)

related:
https://github.com/webaverse/app/issues/3531

result:

https://user-images.githubusercontent.com/60634884/183294177-88036462-56e6-430d-adb5-e37f0f6d7f66.mp4

### CODE REVIEW

File: character-sfx.js [hunk 1]
```diff
@@ -86,24 +86,25 @@ class CharacterSfx {
...
     this.currentStep = null;
     this.currentSwimmingHand = null;
     this.setSwimmingHand = true;
+    this.setEquipSound = false;
   }
   update(timestamp, timeDiffS) {
     if (!this.player.avatar) {
       return;
     }
+    if (this.player.isLocalPlayer && !this.setEquipSound) {
```
Comment (avaer) [inline character-sfx.js:98]: If this is a listener then it should not be listened in the main loop. This belongs in setup code.

**Review by avaer — CHANGES_REQUESTED** [review]
Code in the wrong place.

**Comment by tcm390** [comment]
https://github.com/webaverse/app/pull/3545/files

create character-sfx for localPlayer after `isLocalPlayer` has been set

File: character-sfx.js (hunk 1 above)
Comment (tcm390) [reply character-sfx.js]: https://github.com/webaverse/app/pull/3545/files

create characterSfx for localPlayer after `isLocalPlayer` has been set (like characterPhysics)

**Comment by avaer** [comment]
How about we instead ensure that isLocalPlayer is initialized early enough? it could be an argument to `PlayerBase` `constructor`.

**Review by avaer — CHANGES_REQUESTED** [review]
Recommending refactor.

**Comment by alisaad673** [comment]
Works Fine

File: character-controller.js [hunk 2]
```diff
@@ -1040,7 +1039,8 @@ class LocalPlayer extends UninterpolatedPlayer {
     this.isLocalPlayer = !opts.npc;
     this.isNpcPlayer = !!opts.npc;
     this.detached = !!opts.detached;
-
+    
+    this.characterSfx = new CharacterSfx(this);
```
Comment (avaer) [inline character-controller.js:1043]: Both of these should be put in PlayerBase.

**Review by avaer — CHANGES_REQUESTED** [review]
Still have initialization code in the wrong classes.

File: character-controller.js (hunk 2 above)
Comment (avaer) [reply character-controller.js:1043]: Especially the character sfx, since that should affect remote players too.

**Pull request closed without merging**